openff/toolkit/topology/molecule.py

```diff
@@ -1661,23 +1640,12 @@ def to_inchi(self, fixed_hydrogens=False, toolkit_registry=GLOBAL_TOOLKIT_REGIST
              If an invalid object is passed as the toolkit_registry parameter
         """
 
-        if isinstance(toolkit_registry, ToolkitRegistry):
-            inchi = toolkit_registry.call(
-                "to_inchi", self, fixed_hydrogens=fixed_hydrogens
-            )
-        elif isinstance(toolkit_registry, ToolkitWrapper):
-            toolkit = toolkit_registry
-            inchi = toolkit.to_inchi(self, fixed_hydrogens=fixed_hydrogens)
-        else:
-            raise InvalidToolkitRegistryError(
-                "Invalid toolkit_registry passed to to_inchi. Expected ToolkitRegistry or ToolkitWrapper. "
-                f"Got {type(toolkit_registry)}"
-            )
+        toolkit_registry = _ensure_toolkit_registry(toolkit_registry)
 
-        return inchi
+        return toolkit_registry.call("to_inchi", self, fixed_hydrogens=fixed_hydrogens)
```

Comment:
This pattern could be further golfed down to a one-liner


```
return _ensure_toolkit_registry(toolkit_registry)..call("method_name", ...)
```
but I think that's too far

Reply:
Haha yeah, it's already golfy enough :-P